src/app/listener.ts

```diff
@@ -25,19 +26,27 @@ export class Listener {
       if (message.author.bot) {
         return;
       }
-      try {
-        this._handlers.forEach(async (handler: IHandler) => {
-          await handler.execute(message);
-        });
-      } catch (err) {
-        console.error(err);
-      }
+      this._executeHandlers(this._messageHandlers, message);
     });
   }
 
   private _initializeHandlers(): void {
-    this.container.handlerService.handlers.forEach((Handler) => {
-      this._handlers.push(new Handler(this.container));
+    this.container.handlerService.messageHandlers.forEach((Handler) => {
+      this._messageHandlers.push(new Handler(this.container));
+    });
+
+    this.container.handlerService.channelHandlers.forEach((Handler) => {
+      this._channelHandlers.push(new Handler(this.container));
+    });
+  }
+
+  private async _executeHandlers(handlers: IHandler[], message?: IMessage) {
+    handlers.forEach(async (handler: IHandler) => {
+      try {
+        await handler.execute(message);
+      } catch (e) {
+        console.error(e);
+      }
```

Comment:
why not just let Promises do their thing here?

Reply:
we shud still await here since im not calling await on `_executeHandlers` i think